### ISSUE TITLE
[BUG] Scaling issue with Llama 2 models? Vicuna 13B v1.5 16K in particular!

### ISSUE THREAD

**Issue by StefanDanielSchwarz** [LostRuins/koboldcpp#378]
# Prerequisites

Please answer the following questions for yourself before submitting an issue.

- [x] I am running the latest code. Development is very rapid so there are no tagged versions as of now.
- [x] I carefully followed the [README.md](https://github.com/ggerganov/llama.cpp/blob/master/README.md).
- [x] I [searched using keywords relevant to my issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/filtering-and-searching-issues-and-pull-requests) to make sure that I am creating a new issue that is not already open (or closed).
- [x] I reviewed the [Discussions](https://github.com/ggerganov/llama.cpp/discussions), and have a new bug or useful enhancement to share.

# Expected Behavior

The Llama 2-based model vicuna-13B-v1.5-16K ([TheBloke/vicuna-13B-v1.5-16K-GGML](https://huggingface.co/TheBloke/vicuna-13B-v1.5-16K-GGML]) uses linear RoPE scaling and should work properly with 16K context using the command line arguments `--contextsize 16384 --ropeconfig 0.25 10000`.

# Current Behavior

Tested with Emily in Kobold Lite UI, context size set accordingly:

`--contextsize 16384 --ropeconfig 0.25 10000` exhibits confusion from fourth message on, talking as the user, mixing up punctuation, as if the model is dumber than it should be.

`--contextsize 4096 --ropeconfig 1 10000` doesn't work at all and outputs gibberish.

`--contextsize 4096` exhibits repetition/looping right from the start, after just three messages.

`--contextsize 16384` gives me the best results (although it shouldn't?) - but breaks down after 4K context with repetition/looping.

# Environment and Context

koboldcpp-1.40.1 on Windows 11

# Failure Information (for bugs)

Are we really sure that the whole scaling stuff is working as intended?

Considering the repetition/looping issues that plague most Llama 2 models, I'm wondering if the scaling handling might be involved there as well - as Llama 1 models aren't affected, and they work at native 2K context. For Llama 2 models, we need to use `--contextsize 4096 --ropeconfig 1 10000` for native 4K context, but we usually hit the repetition/looping issues after 2K-3K tokens. And with the 16K model tested here, the discrepancy seems even more apparent.

# Steps to Reproduce

Please provide detailed steps for reproducing the issue. We are not sitting in front of your screen, so the more detail the better.

1. Use different scaling settings
2. Talk to Emily for a couple of messages
3. See which scaling settings gives better or worse results
4. Consider if that makes sense or is unexpected

**Comment by SabinStargem** [comment]
Dunno if it will work for your model, but try these at 16384 context.  IIRC, they work for Airoboros 33b 16k.

0.5 70000
0.25 82000

**Comment by LostRuins** [comment]
The whole scaling stuff is definitely "working" but results may vary based on what the model was fine tuned on, thats why you're able to set your own values for the rope frequency base and scale. Have you tried `--ropeconfig 0.5 10000`?

**Comment by Tacx79** [comment]
I'm at 1.39.1 version and I use q6k version of this model with 16k context and 0.25 10000, outputs seem fine when I throw at it ~10k context but I also use mirostat sampling with it.

I think without mirostat output wasn't so good but with it I'm comfortable enough to use this model interchangeably with 30b instruct llama (without mirostat) when I need longer context.

**Comment by SabinStargem** [comment]
I have been working with Airoboros 33b 16k today, seems that the ROPE settings work differently now that Kobold does 16k.  Try [0.125 3000], with Mirostat, Creative, and Shortwave.   They work in Silly Tavern.  Hopefully my results can work for other 16k models and people.

**Comment by StefanDanielSchwarz** [comment]
Just wanted to report back my findings after much more experimentation:

`--contextsize 16384 --ropeconfig 0.125 20000` didn't work well for me, it mixed markdown code into the conversation.

But the officially recommended `--contextsize 16384 --ropeconfig 0.25 10000` works very well for me now, after I changed my repetition settings (of all things).

My old repetition settings were Rep. Pen. 1.1, Rep. Pen. Range 2048, Rep. Pen. Slope 0.2. **My new repetition settings are Rep. Pen. 1.18, Rep. Pen. Range 2048, Rep. Pen. Slope 0.** Other settings stayed Deterministic: Temp. 0, top_p 0, top_k 1, top_a 0, typical 1, tfs 1, no mirostat.

With those settings, I can no longer reproduce the confusion, talking as the user, and mixing up punctuation issues I used to have. It feels like a new, better model this way. In fact, Vicuna 13B v1.5 16K is now one of my favorites, thanks to the highly extended context and its ability to handle even very complex character cards in SillyTavern (went all the way to 16K max context in over 90 messages during my last test - no issues at all, it was an excellent conversation/roleplay).

The adjusted repetition penalty settings also fixed issues with other models I experienced recently, like MythoMax-L2-13B talking/acting as the user from the start, so it's not just Vicuna 13B v1.5 16K that benefited from the change.

YMMV, of course, but I wanted to report back my latest experiences and conclusions. I'll close this issue since I don't think anymore that the scaling itself is an issue.
